libkbfs/folder_branch_ops.go

```diff
@@ -2494,16 +2500,82 @@ func (fbo *folderBranchOps) createEntryLocked(
 		newBlock = &FileBlock{}
 	}
 
-	de, err := fbo.syncBlockAndFinalizeLocked(
-		ctx, lState, md, newBlock, dirPath, name, entryType,
-		true, true, zeroPtr, excl)
+	// Cache update and operations until batch happens.  Make a new
+	// temporary ID and directory entry.
+	newID, err := fbo.config.cryptoPure().MakeTemporaryBlockID()
+	if err != nil {
+		return nil, DirEntry{}, err
+	}
+	session, err := fbo.config.KBPKI().GetCurrentSession(ctx)
 	if err != nil {
 		return nil, DirEntry{}, err
 	}
-	node, err := fbo.nodeCache.GetOrCreate(de.BlockPointer, name, dir)
+
+	newPtr := BlockPointer{
+		ID:         newID,
+		KeyGen:     md.LatestKeyGeneration(),
+		DataVer:    fbo.config.DataVersion(),
+		DirectType: DirectBlock,
+		Context: kbfsblock.MakeFirstContext(
+			session.UID, keybase1.BlockType_DATA),
+	}
+	node, err := fbo.nodeCache.GetOrCreate(newPtr, name, dir)
 	if err != nil {
 		return nil, DirEntry{}, err
 	}
+	de := DirEntry{
+		BlockInfo: BlockInfo{
+			BlockPointer: newPtr,
+			EncodedSize:  0,
+		},
+		EntryInfo: EntryInfo{
+			Type: entryType,
+			Size: 0,
+		},
+	}
+	fbo.blocks.AddDirEntryInCache(lState, dirPath, name, de)
+	fbo.dirOps = append(fbo.dirOps, cachedDirOp{co, []Node{dir, node}})
+
+	cleanupFn := func() {
+		// Until KBFS-2076 is done, clear out the cached dir data manually.
+		fbo.dirOps = nil
```

Comment:
So for now, you're wiping `dirOps` after every operation. I assume that cleanup will go away, and the batching will take care of that eventually?